05.wc/lib/wc.rb

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+def wc_output
+  opt = OptionParser.new
+  wc_options = { lines: false, words: false, bytes: false }
+  opt.on('-l') { |v| wc_options[:lines] = v }
+  opt.on('-w') { |v| wc_options[:words] = v }
+  opt.on('-c') { |v| wc_options[:bytes] = v }
+  opt.parse!(ARGV)
+
+  counts_by_file = read_files
+  body = format_counts(counts_by_file, wc_options)
+  total = format_total(counts_by_file, wc_options) if counts_by_file.size > 1
+  [body, total].join
+end
+
+def read_files
+  counts = []
+  l = 0
+  w = 0
+  c = 0
+  ARGF.each do |line|
+    l += 1
+    w += line.split.size
+    c += line.bytesize
+    if ARGF.file.eof?
+      counts << { lines: l, words: w, bytes: c, path: $stdin.tty? ? ARGF.file.path : nil }
+      l = 0
+      w = 0
+      c = 0
+    end
+  end
+  counts
+end
+
+def format_counts(counts, wc_options)
+  output = []
+  counts.each do |count|
+    wc_options.each_key do |option|
+      output << format_as_tab(count[option]) if selected_option?(wc_options, option)
```

Comment:
```suggestion
    wc_options.each do |key, flag|
      output << format_as_tab(count[key]) if flag || wc_options.values.none?
```

みたいにした方がややシンプルかも。

ちなみに `wc_options.values.none?` はループのたびに毎回同じ値を返すはずなので、処理効率を考えるとそれよりも・・・以下は宿題ですw

Reply:
ご指摘の通り、ブロック引数の見直しと同じ値はループの外に出すように修正しました。
fdbe6d7b735f696f8b335e5a9d9570a55e424a9b